src/Decapodes.jl

```diff
@@ -17,8 +17,9 @@ using DiagrammaticEquations.Deca
 
 export
 findname, flat_op,
-gensim, evalsim, closest_point, findnode, compile, compile_env,  PhysicsState,  default_dec_matrix_generate, default_dec_generate, default_dec_generate, VectorForm,
-CartesianPoint, SpherePoint, r, theta, phi, TangentBasis, θhat, ϕhat  
+gensim, evalsim, closest_point, findnode, compile, compile_env,  PhysicsState,  default_dec_matrix_generate, default_dec_cu_matrix_generate, default_dec_generate, VectorForm,
+CartesianPoint, SpherePoint, r, theta, phi, TangentBasis, θhat, ϕhat,
+cpu, cuda
```

Comment:
Pick better names for these types

Reply:
I think these types are nice and they match with similar types used for device selection in something like PyTorch. Did you have anything else in mind?

Reply:
Types should at least have capital names like CPU and CUDA

Reply:
Originally those were the names I was using but CUDA name conflicts with the package name, also CUDA, so that won't work well. I could change it to be ```Cpu``` and ```Cuda```.

Reply:
Can have an abstract type for CompilationTarget and subtype CUDATarget CPUTarget

Reply:
I already do that here: https://github.com/AlgebraicJulia/Decapodes.jl/blob/6ae30c6efecea9b7c6b536bc9719368a735255ab/src/simulation.jl#L14-L17